velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -0,0 +1,294 @@
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemPlanNodeId;
+
+  const auto stage1 =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              selectedRowType,
+              makeTableHandle(std::move(filters)),
+              allRegularColumns(selectedRowType))
+          .capturePlanNodeId(lineitemPlanNodeId)
+          .project(std::move(projectAlias))
+          .project(
+              {"l_returnflag",
+               "l_linestatus",
+               "l_quantity",
+               "l_extendedprice",
+               "l_extendedprice * (1.0 - l_discount) AS l_sum_disc_price",
+               "l_extendedprice * (1.0 - l_discount) * (1.0 + l_tax) AS l_sum_charge",
+               "l_discount"})
+          .partialAggregation(
+              {0, 1},
+              {"sum(l_quantity)",
+               "sum(l_extendedprice)",
+               "sum(l_sum_disc_price)",
+               "sum(l_sum_charge)",
+               "avg(l_quantity)",
+               "avg(l_extendedprice)",
+               "avg(l_discount)",
+               "count(0)"})
+          .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .localPartition({}, {stage1})
+          .finalAggregation(
+              {0, 1},
+              {"sum(a0)",
+               "sum(a1)",
+               "sum(a2)",
+               "sum(a3)",
+               "avg(a4)",
+               "avg(a5)",
+               "avg(a6)",
+               "count(a7)"},
+              {DOUBLE(),
+               DOUBLE(),
+               DOUBLE(),
+               DOUBLE(),
+               DOUBLE(),
+               DOUBLE(),
+               DOUBLE(),
+               BIGINT()})
+          .orderBy({0, 1}, {core::kAscNullsLast, core::kAscNullsLast}, false)
+          .planNode();
+
+  TpchPlan context;
+  context.plan = std::move(plan);
+  context.dataFiles[lineitemPlanNodeId] = getTableFilePaths(kTableName);
+  context.dataFileFormat = format_;
+  return context;
+}
+
+TpchPlan TpchQueryBuilder::getQ6Plan() const {
+  const std::string kTableName = "lineitem";
+  std::vector<std::string> selectedColumns = {
+      "l_shipdate", "l_extendedprice", "l_quantity", "l_discount"};
+
+  // Get the corresponding selected column names in the file.
+  const auto aliasSelectedColumns =
+      getColumnAliases(kTableName, selectedColumns);
+
+  auto columnSelector =
+      std::make_shared<facebook::velox::dwio::common::ColumnSelector>(
+          tableMetadata_.at(kTableName).type, aliasSelectedColumns);
+  auto selectedRowType = columnSelector->buildSelectedReordered();
+  // We need to add an extra project step to map file columns to standard
+  // column names.
+  const auto projectAlias =
+      getProjectColumnAliases(kTableName, selectedColumns);
+
+  auto shipDate = getColumnAlias(kTableName, "l_shipdate");
+  common::test::SubfieldFiltersBuilder filtersBuilder;
+  // DWRF does not support Date type. It uses Varchar instead.
```

Comment:
This copy is duplicated in q1 as well. Any chance it can be refactored into a helper method?

Reply:
Both filters are different. Can't think of a helper method.